commons/src/main/java/com/powsybl/commons/reporter/TreeReporterSerializer.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.reporter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class TreeReporterSerializer extends StdSerializer<TreeReporter> {
+
+    private static final String VERSION = "1.0";
+    private final Map<String, String> dictionary;
```

Comment:
Ok, thanks for explaining! Additional context is really needed here, so I'll see how to pass it through `DeserializationContext`.

Reply:
The serializers/deserializers are now stateless. And in the end I managed to avoid that need of additional context, by directly calling methods recursively when serializing/deserializing instead of calling the serializer/deserializer itself.

Reply:
Yes very good idea, we will need the jackson module too.

Reply:
Done!